src/main/java/org/opensearch/knn/training/TrainingJobRunner.java

```diff
@@ -176,4 +193,122 @@ private void serializeModel(TrainingJob trainingJob, ActionListener<IndexRespons
     public int getJobCount() {
         return jobCount.get();
     }
+
+    /**
+     * This method is called whenever the cluster state changes. It is used to update models that are still training when a node leaves or the cluster crashes.
+     * @param event the event that changed the cluster change
+     */
+    @Override
+    public void clusterChanged(ClusterChangedEvent event) {
+        if (event.localNodeClusterManager()) {
+            if (event.isNewCluster()) {
+                threadPool.schedule(() -> {
+                    try {
+                        updateModelsNewCluster();
+                    } catch (IOException | InterruptedException | ExecutionException e) {
+                        throw new RuntimeException(e);
+                    }
+                }, TimeValue.timeValueSeconds(1), ThreadPool.Names.GENERIC);
```

Comment:
Why 1 second here and zero second in line#220?

Reply:
The 1 second occurs on cluster startup because the modelDao object has not been created yet.  So we need to schedule it for after the object is created.  The 0 second occurs while the cluster is running, so modelDao has already been created.  However, the schedule is still needed so that it can run on a separate thread and not crash the cluster